controller/api/destination/watcher/ip_watcher.go

```diff
@@ -123,61 +123,49 @@ func (iw *IPWatcher) Unsubscribe(clusterIP string, port Port, listener EndpointU
 // GetSvcID returns the service that corresponds to a Cluster IP address if one
 // exists.
 func (iw *IPWatcher) GetSvcID(clusterIP string) (*ServiceID, error) {
-	resource, err := getResource(clusterIP, iw.k8sAPI.Svc().Informer())
+	objs, err := iw.k8sAPI.Svc().Informer().GetIndexer().ByIndex(podIPIndex, clusterIP)
 	if err != nil {
-		return nil, err
+		return nil, status.Error(codes.Unknown, err.Error())
 	}
-	if svc, ok := resource.(*corev1.Service); ok {
-		service := &ServiceID{
-			Namespace: svc.Namespace,
-			Name:      svc.Name,
-		}
-		return service, nil
+	services := make([]*corev1.Service, 0)
+	for _, obj := range objs {
+		service := obj.(*corev1.Service)
+		services = append(services, service)
 	}
-	// Either `clusterIP` does not map to a service that the indexer is aware
-	// of, or it maps to a resource that is ignored
-	return nil, nil
-}
-
-// GetPod returns the pod that corresponds to an IP address if one exists.
-func (iw *IPWatcher) GetPod(podIP string) (*corev1.Pod, error) {
-	resource, err := getResource(podIP, iw.k8sAPI.Pod().Informer())
-	if err != nil {
-		return nil, err
+	if len(services) > 1 {
+		return nil, status.Errorf(codes.FailedPrecondition, "Cluster IP address conflict: %s/%s, %s/%s", services[0].Namespace, services[0].Name, services[1].Namespace, services[1].Name)
```

Comment:
How do we know there are only 2 always? Why not `string,Join` all of them and use that string 🤔

Reply:
There could be more than two in this case! I just kept this error as is though, and it has only ever printed the first two conflicting pods.

It could be changed, but it's not related to this fix so I'd rather keep it as is.

Reply:
Yeah, With the current hostIP issue, There can be more than two in most cases.

If we are going to print those names, My suggestion would be to print them all unless if there is a reason not to. :)

Reply:
I'll update this error message to provide a little more description. I don't want to list all pods because in an extreme case we could end up hitting the message limit again. I will however change it to include the number of conflicting pods:

```
Cluster IP address conflict between %d pods; first two: ...
```

That way a user can see they have for example 10 conflicting pods, and the first two give them a starting place to look.